### PR TITLE
Fix imported micro xs when lower energy grid bound is higher than upper bound

### PR DESCRIPTION
**Pull request by amandalund**
It’s possible for the lower energy bound of the microscopic cross section grid to be higher than the upper bound if the production cut values are very high. Simply skip building the grid if this is the case instead of requiring `min_energy < max_energy`.

### CODE REVIEW

File: src/celeritas/ext/detail/GeantModelImporter.cc [hunk 1]
```diff
@@ -146,10 +146,12 @@ ImportModel GeantModelImporter::operator()(G4VEmModel const& model) const
...
         double max_energy = model.HighEnergyLimit() / CLHEP::MeV;
         CELER_ASSERT(0 <= min_energy);
-        CELER_ASSERT(min_energy < max_energy);
 
+        // It's possible for the lower energy bound to be higher than the upper
+        // energy bound if the production cut values are very high. Don't build
+        // the xs grid if this is the case.
         auto& model_mat = result.materials[mat_idx];
-        if (needs_micro_xs(result.model_class))
+        if (needs_micro_xs(result.model_class) && min_energy < max_energy)
```
Comment (whokion) [inline src/celeritas/ext/detail/GeantModelImporter.cc:154]:
```suggestion
        bool export_micros = needs_micro_xs(result.model_class);
        if (CELER_UNLIKELY(!(min_energy < max_energy)))
        {
          CELER_LOG(info) << "Minimum energy " << min_energy
                               << " [MeV] is not less than the maximum energy "
                               << max_energy << " [MeV] for model '"
                               << model.GetName() << "' in material '"
                               << g4mat.GetName() << "' (low energy limit is "
                               << model.LowEnergyLimit() / CLHEP::MeV
                               << " and minimum primary energy for cutoff "
                               << cutoff << " is " << min_primary;
            export_micros = false;
        }
       if (export_micros)       
```

Comment (whokion) [reply src/celeritas/ext/detail/GeantModelImporter.cc:154]: This was the patch code from @sethrj

**Review by whokion — APPROVED** [review]
Your change is essentially equivalent to the suggested code, which was from @sethrj.  Nevertheless, I still do not understand why CMS imposes such a cut for those materials in the `RP_NonDenseMaterialsCuts `region which we need to find out.  Let's merge this unless @sethrj has a different thought or opinion.

File: src/celeritas/ext/detail/GeantModelImporter.cc (hunk 1 above)
Comment (amandalund) [reply src/celeritas/ext/detail/GeantModelImporter.cc]: Added some debug logging.

**Review by sethrj — COMMENTED** [review]
This looks good to me, but do we have to go further? What do the macro cross sections (physics tables) look like for this case? Could you perhaps add a tiny `.gdml` file with a 50m production cut and a `GeantImporter` test, and see if we can set up the physics correctly?

**Comment by sethrj** [comment]
Original reported issue (by @whokion) for provenance:

---

> Following up the issue at `GeantModelImporter.cc159` (min_energy= 9549.65 max_energy= 1000) from CMSSW, I find that demo-geant-integration fails with both cms-run3.gdml and cms2018.gdml (under benchmarks/full-cms-test/input/).
> Failure by running `demo-geant-integration simple-cms-test.mac` by replacing the simple-cms.gdml
> 1) with `cms-run3.gdml`
```
 issued by : celeritas/ext/detail/GeantModelImporter.cc:149
 internal assertion failed: min_energy < max_energy 
```
> 2) with `cms-2018,gdml`
```
celeritas/geo/GeoMaterialParams.cc:154: warning: Some geometry volumes do not have known material IDs: CMStoZDC0x7f4a9a83fd80@0x24cadb0
```

---

@amandalund debugged:

> looking into that high min_energy, I see several materials with a 50 m range cut (Vacuum, PPS_AISI-316L-Steel, Air, Diamond, Gold) which translates into the very high energy production cut/min energy. At the bottom of the cms-run3.gdml there is some auxiliary information where cuts are set for regions — the `RP_NonDenseMaterialsCuts` region has a 50 m range cut for all particle types, and I think all those materials are in that region.

**Comment by amandalund** [comment]
> This looks good to me, but do we have to go further? What do the macro cross sections (physics tables) look like for this case? Could you perhaps add a tiny `.gdml` file with a 50m production cut and a `GeantImporter` test, and see if we can set up the physics correctly?
> 
> We might also want to add a stepper test to see if we can track through it correctly.

Good point, will do!

**Review by sethrj — APPROVED** [review]

**Pull request merged**